src/core.js

```diff
@@ -1593,8 +1614,16 @@ function _buildSquel(flavour = null) {
       }));
     }
 
-    having (condition, ...values) {
-      this._condition(condition, ...values);
+    having (field, operator, ...values) {
+      let validOperators = ['=', '<', '>', '<=', '>=', '<>', '!=', 'in', 'not in', 'like', 'not like', 'is', 'is not'];
```

Comment:
Thanks for looking into this. Yes, I'm aware of the duplication and I agree. I have some improvements planned for defining and checking the valid operators. I will submit them in another commit soon. The duplicate code will be gone then and I will move the code blocks to the appropriate base classes as you suggested.